doc/design/model_zoo.md

```diff
@@ -0,0 +1,174 @@
+
+```sql
+SELECT ... TO PREDICT ... USING my_first_model
+```
+
+If another analyst wants to use the trained model, he would need to use the full name as `an_analyst/my_first_model`.
+
+```sql
+SELECT ... TO PREDICT ... USING an_analyst/my_first_model
+```
+
+There could be more than one model definitions in each model's Docker image. We need to be able to find out which model definition is used to train current saved model. Also if we only want to reuse the model definition to train a new model we need to know the model definition class name, so that we can pass it to the `TRAIN` clause.
+
+To list all trained models of one user, you can do:
+
+```sql
+SQLFLOW LIST an_analyst
+```
+
+This should output a table showing the saved models and which model definition it was using.
+
+```
+|  saved model    |   model def    |
+| my_first_model  | MyDNNRegressor |
+| my_second_model | MyDNNRegressor |
+```
+
+Within **any** deployment that have internet access, to list published models:
+
+```sql
+SQLFLOW LIST [models.sqlflow.com/an_analyst]
+```
+
+Display model definitions and documentation of the published model:
+
+```
+SQLFLOW DESCRIBE model.sqlflow.com/an_analyst/my_first_model;
+| available model defs |
+| MyDNNRegressor       |
+| MyDNNClassifier      |
+
+SQLFLOW DESCRIBE model.sqlflow.com/an_analyst/my_first_model.MyDNNRegressor;
+
+Documatation for my_first_model.MyDNNRegressor
+...
+...
+```
+
+Use a published model to predict some data:
+
+```sql
+SELECT ... TO PREDICT employee.predicted_salary USING model.sqlflow.com/an_analyst/my_first_model
+```
+
+### Model Publication
+
+Publishing a model definition, we hope all users of all SQLFlow deployments can train the model using their datasets. To achieve this goal, data scientists need to set up a continuous deployment system to automatically build their model definition repositories into Docker images and push to a public Docker registry like DockerHub.com.  In the above example, the Docker image `a_data_scientist/regressors` is one hosted on DockerHub.com.
+
+It requires more steps to publish a trained model. We need a public registry like DockerHub.com, but hosts rows from the model zoo table and corresponding model parameters.  We propose the following extended SQL syntax for analysts `an_analyst` to publish her trained model.
+
+```sql
+SQLFLOW PUBLISH my_first_model
+    [TO https://models.sqlflow.com/user_name]
+```
+
+This statement uploads the model parameters and other information to the registry service, which defaults to https://models.sqlflow.com, and under the account `user_name`, which defaults to `an_analyst` in the above example.
+
+Then, another analyst should be able to use the trained model by referring to it in its full name.
+
+```sql
+SELECT ... TO PREDICT employee.predicted_salary USING model.sqlflow.com/an_analyst/my_first_model
```

Comment:
We should use `sqlflow.org`?

Reply:
I am afraid that the model service would be a commercialized one; as we need fundings for storage and computation resource.